Resources/Catalog/FileCatalogFactory.py

```diff
@@ -13,41 +13,52 @@ class FileCatalogFactory:
 
   def __init__( self ):
     self.log = gLogger.getSubLogger( 'FileCatalogFactory' )
+    self.catalogPath = ''
 
   def createCatalog( self, catalogName, useProxy = False ):
     """ Create a file catalog object from its name and CS description
     """
+    catalogPath = getCatalogPath( catalogName )
+    catalogType = gConfig.getValue( catalogPath + '/CatalogType', catalogName )
+    catalogURL = gConfig.getValue( catalogPath + '/CatalogURL', "DataManagement/" + catalogType )
+    optionsDict = {}
+    result = gConfig.getOptionsDict( catalogPath )
+    if result['OK']:
+      optionsDict = result['Value']
+
     if useProxy:
```

Comment:
If I understand correctly, the FileCatalogProxy just forwards the calls to the DFC right ? In that case, I would just import the FileCatalogClient into the FileCatalogProxyClient and have getInterfaceMethods return FileCatalogClient.getInterfaceMethods. It avoids this hack here, and you don't need the setter over which you and Federico are debatting